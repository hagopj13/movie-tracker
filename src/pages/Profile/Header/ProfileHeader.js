// @flow
import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import { blueGrey } from '@material-ui/core/colors';

import profileSelectors from 'store/user/profile/profile.selectors';

type Props = {
  userName: string,
};

const useStyles = makeStyles((theme) => ({
  root: {
    background: `linear-gradient(${blueGrey[700]}, ${blueGrey[600]} 70%, ${blueGrey[500]})`,
  },
  container: {
    padding: theme.spacing(5),
    display: 'flex',
    alignItems: 'center',
    color: theme.palette.common.white,
    [theme.breakpoints.down('xs')]: {
      flexDirection: 'column',
    },
  },
  avatar: {
    fontSize: 40,
    width: theme.spacing(11),
    height: theme.spacing(11),
    backgroundColor: theme.palette.primary.main,
    textTransform: 'uppercase',
    marginRight: theme.spacing(4),
    [theme.breakpoints.down('xs')]: {
      marginRight: 0,
      marginBottom: theme.spacing(2),
    },
  },
}));

export const ProfileHeader = (props: Props) => {
  const { userName } = props;

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Container className={classes.container}>
        <Avatar className={classes.avatar} color="primary">
          {userName?.length > 0 ? userName.charAt(0) : ''}
        </Avatar>
        <Typography className={classes.userName} variant="h4">
          {userName}
        </Typography>
      </Container>
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  userName: profileSelectors.selectUserName,
});

export default connect(mapStateToProps)(ProfileHeader);
