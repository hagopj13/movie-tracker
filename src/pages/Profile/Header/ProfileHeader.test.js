import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import Avatar from '@material-ui/core/Avatar';
import Typography from '@material-ui/core/Typography';

import { ProfileHeader } from './ProfileHeader';

describe('ProfileHeader component', () => {
  let shallow;
  let userName;
  let wrapper;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    userName = 'Some user name';
    const mockProps = {
      userName,
    };
    wrapper = shallow(<ProfileHeader {...mockProps} />);
  });

  it('should render the ProfileHeader component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the first letter of the user name in an Avatar', () => {
    expect(wrapper.find(Avatar).text()).toBe(userName.charAt(0).toUpperCase());
  });

  it('should render the user name', () => {
    expect(wrapper.find(Typography).text()).toBe(userName);
  });
});
