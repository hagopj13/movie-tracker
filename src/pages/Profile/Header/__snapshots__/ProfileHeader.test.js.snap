// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHeader component should render the ProfileHeader component correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-3"
      color="primary"
    >
      S
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      variant="h4"
    >
      Some user name
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(Container))>
</div>
`;
